ttnn/cpp/ttnn/operations/full_like/device/full_like_factory.cpp

```diff
@@ -0,0 +1,132 @@
+    std::vector<uint32_t> reader_compile_time_args = {
+        (uint32_t)src_is_dram,
+    };
+    tt::tt_metal::CircularBufferConfig cb_value_config =
+        tt::tt_metal::CircularBufferConfig(single_tile_size, {{cb_fill_value_id, data_format}})
+            .set_page_size(cb_fill_value_id, single_tile_size);
+    auto cb_fill_value = tt::tt_metal::CreateCircularBuffer(program, all_cores, cb_value_config);
+    std::map<string, string> reader_defines;
+
+    switch (dtype) {
+        case DataType::BFLOAT16: reader_defines["OUTPUT_DTYPE_BFLOAT16"] = "1"; break;
+        case DataType::INT32: reader_defines["OUTPUT_DTYPE_INT32"] = "1"; break;
+        case DataType::FLOAT32: reader_defines["OUTPUT_DTYPE_FLOAT32"] = "1"; break;
+        default: break;
+    }
+
+    bool dst_is_dram = output.buffer()->buffer_type() == tt::tt_metal::BufferType::DRAM ? 1 : 0;
+    std::vector<uint32_t> writer_compile_time_args = {(std::uint32_t)dst_is_dram};
+
+    auto reader_id = tt::tt_metal::CreateKernel(
+        program,
+        "ttnn/cpp/ttnn/operations/full_like/device/kernels/reader_full_like.cpp",
+        all_cores,
+        tt::tt_metal::ReaderDataMovementConfig(reader_compile_time_args, reader_defines));
+
+    auto writer_id = tt::tt_metal::CreateKernel(
+        program,
+        "ttnn/cpp/ttnn/operations/full_like/device/kernels/writer_full_like.cpp",
+        all_cores,
+        tt::tt_metal::WriterDataMovementConfig(writer_compile_time_args));
+
+    uint32_t tiles_offset = 0;
+    for (uint32_t i = 0; i < num_cores; i++) {
+        const CoreCoord core(i / num_cores_y, i % num_cores_y);
+
+        uint32_t num_tiles_per_core = 0;
+        if (core_group_1.core_coord_in_core_ranges(core)) {
+            num_tiles_per_core = num_tiles_per_core_group_1;
+        } else if (core_group_2.core_coord_in_core_ranges(core)) {
+            num_tiles_per_core = num_tiles_per_core_group_2;
+        } else {
+            TT_ASSERT(false, "Core not in specified core ranges");
+        }
+
+        SetRuntimeArgs(program, reader_id, core, {u.u32});
+        SetRuntimeArgs(program, writer_id, core, {output.buffer()->address(), num_tiles_per_core, tiles_offset});
+
+        tiles_offset += num_tiles_per_core;
+    }
+
+    return {std::move(program), {reader_id, writer_id, num_cores, num_cores_y}};
+}
+
+void FullLikeOperation::ProgramFactory::override_runtime_arguments(
+    cached_program_t& cached_program,
+    const operation_attributes_t& operation_attributes,
+    const tensor_args_t& tensor_args,
+    tensor_return_value_t& output) {
+    auto& program = cached_program.program;
+    auto& reader_kernel_id = cached_program.shared_variables.reader_kernel_id;
+    auto& writer_kernel_id = cached_program.shared_variables.writer_kernel_id;
+    auto& num_cores = cached_program.shared_variables.num_cores;
+    auto& num_cores_y = cached_program.shared_variables.num_cores_y;
+    std::cout << "Dit me may" << std::endl;
```

Comment:
```suggestion
    std::cout << "Dit me may" << std::endl;
```
Please remove the debugging code.

Reply:
Resolved